Improving helm setup

knelasevero opened the pull request:
Main thing here is to have a way to see if changes to charts are at least valid in the PR. 

I see ./test/run-helm-tests.sh, but I am not sure how this is being called. In the docs it says that this is being used by the CI, but  I don't see it. Can we remove it?

damemi reviewed (commented):
We configure CI through a prow setup in https://github.com/kubernetes/test-infra. The helm test is run [here](https://github.com/kubernetes/test-infra/blob/5ed7f26/config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-master.yaml#L167). 

So, we shouldn't need a new github workflow for this. But there is some work that needs to be done to improve our helm tests as found in https://github.com/kubernetes-sigs/descheduler/issues/829

ingvagabund commented:
@knelasevero https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-master.yaml#L146-L170

ingvagabund commented:
So based on https://github.com/kubernetes-sigs/descheduler/blob/master/charts/descheduler/templates/tests/test-descheduler-pod.yaml#L30 the helm test only checks the descheduler job completed. @damemi I wonder what's the scope of the helm test? To "just" verify the descheduler got "properly" deployed? Does not seem we are testing any functionality here.

knelasevero commented:
@ingvagabund it doesn't test that also... It seems to pass even when some stuff fails.

@damemi I saw that chart releaser was being used here on GitHub actions, and there are quite a bunch of ready to use GitHub actions for helm, so we don't have to reinvent the wheel. I wonder if we can't leave everything helm related on actions, and everything else on prow [?]

knelasevero commented:
@damemi  Makes sense! Thanks for pointing me in the direction of the prow test :)

knelasevero commented:
Ok, I was having a look over other sig sponsored projects that have anything with helm charts, and it seems that people only check lint and sometimes the ct install. Some use gh-actions, some use prow (most just check linting):

- ingress nginx on prow, but only linting: https://github.com/kubernetes/test-infra/blob/5ed7f2609f61a8ddfb5ee2ecb51f098e6a895edc/config/jobs/kubernetes/sig-network/ingress-nginx/ingress-nginx-presubmit.yaml#L93

- external-dns on gh-actions (same approach that I have added here, and same that we have added for [external-secrets](https://github.com/external-secrets/external-secrets/blob/main/.github/workflows/helm.yml)): https://github.com/kubernetes-sigs/external-dns/blob/master/.github/workflows/lint-test-chart.yaml

I think it would make sense to remove the one from prow and leave it here. Would require a PR to test-infra, and then I would remote the script from here. So let me know if I should pursue that.

damemi commented:
@knelasevero sounds good to me, just wanted to make sure we weren't doing something totally unprecedented. Thanks for offering to handle the work for that

knelasevero commented:
Started by removing the test from prow: https://github.com/kubernetes/test-infra/pull/26630

After that I will continue removing stuff from here, and then I will go forward with improving the helm checks on gh-actions.

knelasevero commented:
Hey @damemi, I will remove wip from this PR then. So we can merge the helm workflow and check its results. (as per your comment on https://github.com/kubernetes/test-infra/pull/26630#issuecomment-1161912089).

Then after that I can followup with removals in k/test-infra and also removals of the scripts here.

damemi commented:
/lgtm
